app.js

```diff
@@ -185,7 +186,48 @@ function MonitorApp ({
       });
     };
     
+    // Store in memcache db
+    // TODO: deprecate memcache db. just use mongo for everything.
     mjs.set(key, JSON.stringify(newRoutes), {}, handleErr);
+    
+    // Add new routes to mongo db log. Used for generating timeseries.
+    db.collection('routeLog').insertOne({
```

Comment:
Weird to default `db` to `null` above and then try to use it here, which I believe would throw an exception.

I think it makes sense to ask here 'Is a `db` now required in our interface)

I think it would be more clear to have no `db` default, and to check for `db`'s truthiness before using it. (`if (db) { /* ... */ }`)
Or, just say 'we need a db to do anything' and check at top of function and throw an explicit TypeError if it's not an object.

Reply:
Makes sense. What do you think of the way www instantiates the mongo connection and passes it in? At first I tried letting the app create the connection, but then it seemed kind of lame to try to instantiate an object async-ly.

Reply:
I think what you have done is good, i.e. to allow something to be injected here to manage storage, instead of all that being tightly coupled to the rest of MonitorApp. Ideally that injected storage object would also abstract away various ways of storing things (e.g. a mongo server, in-memory storage, file-based storage, idk).

As implemented, the interface of what is injected in is somewhat tied to MongoDB as a storage engine. As the app grows it may make sense to define a specific [service layer](https://martinfowler.com/eaaCatalog/serviceLayer.html) interface for various ways of updating these routeLog records, call it RouteLogService. You could instantiate everything like
```
const app = MonitorApp({
  routeLogService: MongoRouteLogService({
    db: createMongoDb()
  })
});
```

Usually I'd make a corresponding `InMemoryRouteLogService` for running tests against and stuff. But wohle point is to have MonitorApp depend on an interface that is not coupled to MongoDbClient.